meshtastic/tests/test_node_runtime_transport.py

```
@@ -0,0 +1,1233 @@
+
+        mock_local_node.channels = channels
+
+        def make_cache_none(*args: Any, **kwargs: Any) -> None:
+            mock_local_node.channels = None
+
+        with caplog.at_level(logging.WARNING):
+            with patch.object(
+                delete_channel_runtime._channel_write_runtime,
+                "write_channel_snapshot",
+                side_effect=make_cache_none,
+            ):
```

Ruff (0.15.6)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)